#!/usr/bin/env python
PACKAGE = "mobile_robot_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add("ref_position_x", double_t, 0, "ref position x (m)", 0.0,-100.0,100.0)
#gen.add("ref_position_y", double_t, 0, "ref position y (m)", 0.0,-100.0,100.0)
#gen.add("ref_yaw", double_t, 0, "ref yaw (deg)", 0.0,-180,180)
gen.add("vitesse_max",double_t, 0, "vitesse max (m/s)", 1.0, 0.0, 5.0);
gen.add("vitesse_angulaire__max",double_t,0,"saturation vitesse angulaire (rad/s)",2.0,0.0,6.0);
gen.add("gain_rotation",double_t, 0, "gain rotation", 2.0, 0.0, 10.0);
gen.add("gain_vitesse",double_t,0, "gain vitesse", 1.0, 0.0, 10.0);
gen.add("gain_gradient",double_t,0, "gain gradient", 0.2, 0.0, 10.0);
gen.add("threshold_distance",double_t,0,"threshold distance (m)",0.05,0.0,1.0);
gen.add("threshold_yaw",double_t,0,"threshold yaw (rad)",1.0,0.0,6.0);
gen.add("threshold_waypoints",double_t,0,"threshold waypoints (m)",0.3,0.0,2.0);
gen.add("gain_p_ftv",double_t, 0, "proportional filter v", 0.4, 0.0, 5.0);
gen.add("gain_d_ftv",double_t, 0, "derivative filter v", 2.0, 0.0, 5.0);
gen.add("gain_p_ftw",double_t, 0, "proportional filter w", 0.1, 0.0, 5.0);
gen.add("gain_d_ftw",double_t, 0, "derivative filter w", 1.0, 0.0, 5.0);
gen.add("gain_pe",double_t, 0, "gain PE", 0.1, 0.0, 5.0);
gen.add("frequency_pe",double_t, 0, "Frequency PE", 2, 0.0, 500.0);
gen.add("distance_constraint",bool_t,0,"flag distance constraint",False);
gen.add("fov_constraint",bool_t,0,"flag fov constraint",False);


exit(gen.generate(PACKAGE, "mobile_robot_controller", "MobileRobotController"))
